src/common/storage/src/config.rs

```diff
@@ -149,6 +154,41 @@ impl Default for StorageFsConfig {
     }
 }
 
+pub const STORAGE_FTP_DEFAULT_ENDPOINT: &str = "ftps://127.0.0.1";
+/// Config for FTP and FTPS data source
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct StorageFtpConfig {
+    pub endpoint: String,
+    pub root: String,
+    pub username: String,
+    pub password: String,
+    pub secure: bool,
```

Comment:
Do we need to remove this?

Reply:
I prefer keeping it so that no extra calculation is needed in `StorageParams::is_secure`.

Reply:
> I prefer keeping it so that no extra calculation is needed in `StorageParams::is_secure`.

But this will leak `secure` config to users.

Reply:
How about making it private and add an `fn is_secure(&self) -> bool` to StorageFtpConfig?

Reply:
> How about making it private and add an `fn is_secure(&self) -> bool` method to StorageFtpConfig?

Ok this is a bad idea, I'll just remove it.